types.go

```diff
@@ -109,8 +106,14 @@ func NewSet(s []types.Value) Set {
 	return types.NewSet(s)
 }
 
-// UnsafeDecimal creates a decimal via unsafe conversion from int, int64, float64.
-// Precision may be lost and overflows may occur.
-func UnsafeDecimal[T int | int64 | float64](v T) Decimal {
-	return types.UnsafeDecimal(v)
+func NewDecimal(i int64, exponent int) (Decimal, error) {
```

Comment:
Can you add docs for all these functions?

Reply:
Also, I see the other ParseIP, etc funcs aren't wrapped here, maybe those could be done as well?

Reply:
Good catch. I wish the story for projecting functions from another package was a little bit neater on the Go documentation side.